fms/utils/serialization.py

```diff
@@ -86,8 +91,30 @@ def get_adapted(
 from fms import models
 
 
+def get_ckp_format(model_path: Union[str, Path]) -> str:
+    """
+    Returns the checkpoint format of a model checkpoint. If format
+    is not supported, returns "unk"
+
+    Args:
+    model_path: the path to find the weights.
+    """
+    model_path = Path(os.path.expanduser(model_path))
+    if model_path.suffix == ".pth" or len(sorted(model_path.glob("*.pth"))) > 0:
+        return "pt"
+    if model_path.suffix == ".bin" or len(sorted(model_path.glob("*.bin"))) > 0:
+        return "hf"
+    if (
+        model_path.suffix == ".safetensors"
+        or len(sorted(model_path.glob("*.safetensors"))) > 0
+    ):
+        return "st"
+    return "unk"
+
+
 def load_state_dict(
     model_path: Union[str, Path],
+    checkpoint_format: str,
```

Comment:
as a public function, should we allow None here? users might want to call this separately.

Reply:
I'm also wondering if we should add a `, *,` ... force the optional args to be explicitly specified so we can ensure we don't break calling code with changes.

Reply:
> should we allow None here

I forgot suggesting to make it a required parameter in my previous comments ! :-P ... it does look a bit odd to me now and we could call the function to search for the format again in the None case, in case this was called directly... but... up to you how you want to handle that

Reply:
ok, reworked the API to make it optional again, get the checkpoint format, and make a special "pt_all" format that just gets all files in a folder as PT checkpoints. Let me know what you think